pkg/sql/opt/norm/rules/project_set.opt

```diff
@@ -0,0 +1,27 @@
+# =============================================================================
+# project_set.opt contains normalization rules for the ProjectSet operator.
+# =============================================================================
+
+# ConvertZipArraysToValues applies the unnest zip function to array inputs,
+# converting them into a Values operator within an InnerJoinApply. This allows
+# Values and decorrelation rules to fire. It is especially useful in cases where
+# the array contents are passed as a PREPARE parameter, such as:
+#
+#   SELECT * FROM xy WHERE y IN unnest($1)
+#
+# The replace pattern is equivalent to the match pattern because the
+# InnerJoinApply outputs every value in the array for every row in the input,
+# and outputs nulls to pad shorter arrays. It also supports correlation between
+# the array arguments and the input expression.
+[ConvertZipArraysToValues, Normalize]
+(ProjectSet
+    $input:*
+    $zip:* & (CanConvertToValues $zip)
```

Comment:
Maybe call this `CanConstructValuesFromZips`, and then `ConstructValuesFromZips` below. We'll want to add `jsonb_array_elements` JSON support later on, so this naming will prepare for that. Also, we try to keep our naming symmetric between match and replace patterns in cases like this.